spotlight-client/src/App.tsx

```diff
@@ -15,19 +15,53 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 // =============================================================================
 
+import { RouteComponentProps, Router } from "@reach/router";
 import React from "react";
 import AuthWall from "./AuthWall";
+import PageExplore from "./PageExplore";
+import PageHome from "./PageHome";
+import PageMetric from "./PageMetric";
+import PageNarrativeHome from "./PageNarrativeHome";
+import PageNotFound from "./PageNotFound";
+import PageTenant from "./PageTenant";
+import { DataPortalSlug, NarrativesSlug } from "./routerUtils/types";
+import SiteNavigation from "./SiteNavigation";
 import StoreProvider from "./StoreProvider";
 
+/**
+ * Helps with nesting; all it does is render its children.
+ */
+const PassThroughPage: React.FC<RouteComponentProps> = ({ children }) => (
+  <>{children}</>
+);
+
 const App: React.FC = () => {
   return (
     <StoreProvider>
       <AuthWall>
-        <div>
-          <header>
-            <h1>Spotlight</h1>
-          </header>
-        </div>
+        <SiteNavigation />
+        <main>
+          <Router>
+            {/*
+              NOTE: every leaf route component in this router should be wrapped
```

Comment:
I agree that this required wrapper approach isn't 100% ideal, but with this comment I don't see it as causing problems. What would happen if a leaf is not wrapped in the `withRouterSync`. Can you detect that in the leaf route component and raise a loud error?

Reply:
detecting or requiring the component type seems pretty dicey ... I'm not sure it's even really feasible to do in a way that can't be circumvented by simply not following instructions? Any other solution I can think of would add more complexity or indirection to the router configuration, which doesn't really seem worth it.   

it is technically not required, per se, to use the HOC but there will be some major differences in route behavior which, in combination with this comment, will hopefully be noisy enough to keep our future selves in line:
- components that use the HOC will receive the route params typed as string unions, whereas those that don't will get plain strings. that will make it harder to use the data models which will hopefully be a clue that you have done something wrong
- central state updates that depend on the route parameters will not happen if you don't use the HOC. a given page either won't care about this because it doesn't use any of that data, or it will render the wrong data and hopefully that will be pretty obvious in most cases